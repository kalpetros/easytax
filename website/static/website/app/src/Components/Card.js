import styles from '../../../css/src/Components/Card.css';

import {Link} from 'react-router-dom';
import React from 'react';

class Card extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };

    }
    componentDidMount() {
    }
    render() {
	return(
	    <div className={styles.card}>
              <div className={styles.title}>
                {this.props.title}
              </div>
              <div className={styles.content}>
                {this.props.children}
              </div>
	    </div>
	);
    }
}

export{Card};
