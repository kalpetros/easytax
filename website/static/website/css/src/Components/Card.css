.card {
    background-color: #fff;
    border: 1px solid #eee;
    border-radius: 8px;
    margin: 0.75rem;
    filter: drop-shadow(0 2px 2px #e0e0e0);
}

.title {
    padding: 0.75rem;
    border-bottom: 1px solid #eee;
}

.content {
    padding: 0.75rem;
}
