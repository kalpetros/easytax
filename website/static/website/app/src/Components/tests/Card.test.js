import {Card} from '../Card';

import renderer from 'react-test-renderer';

import React from 'react';

test('Renders a card with just a title', () => {
  const component = renderer.create(
      <Card title="Test"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a card with content', () => {
  const component = renderer.create(
      <Card>Test</Card>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a card with a title and content', () => {
  const component = renderer.create(
      <Card title="Test">Test</Card>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
