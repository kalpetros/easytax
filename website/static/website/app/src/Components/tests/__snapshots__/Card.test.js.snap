// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a card with a title and content 1`] = `
<div
  className="card"
>
  <div
    className="title"
  >
    Test
  </div>
  <div
    className="content"
  >
    Test
  </div>
</div>
`;

exports[`Renders a card with content 1`] = `
<div
  className="card"
>
  <div
    className="title"
  />
  <div
    className="content"
  >
    Test
  </div>
</div>
`;

exports[`Renders a card with just a title 1`] = `
<div
  className="card"
>
  <div
    className="title"
  >
    Test
  </div>
  <div
    className="content"
  />
</div>
`;
